index.js

```diff
@@ -83,9 +83,9 @@ async function fastifyRateLimit (fastify, settings) {
     pluginComponent.store = new Store(globalParams)
   } else {
     if (settings.redis) {
-      pluginComponent.store = new RedisStore(settings.redis, settings.nameSpace || 'fastify-rate-limit-', globalParams.timeWindow, settings.continueExceeding)
+      pluginComponent.store = new RedisStore(settings.redis, globalParams.timeWindow, settings.continueExceeding, settings.nameSpace || 'fastify-rate-limit-')
     } else {
-      pluginComponent.store = new LocalStore(globalParams.timeWindow, settings.cache, fastify, settings.continueExceeding)
+      pluginComponent.store = new LocalStore(settings.cache, globalParams.timeWindow, settings.continueExceeding)
```

Comment:
How does changing these function signatures affect the expectation of user provided `Store` functions?

Reply:
I doesn't because we check for it in the preceding lines and simply pass `globalParams` to a custom Store, LocalStore is ours (in the sense that we implement it), and RedisStore has to be an `ioredis` instance [1]

https://github.com/fastify/fastify-rate-limit/pull/323/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R81-R84

[1] — We mention it in the README as well

> redis: by default this plugins uses an in-memory store, which is fast but if you application works on more than one server it is useless, since the data is stored locally.
You can pass a Redis client here and magically the issue is solved. To achieve the maximum speed, this plugin requires the use of [ioredis](https://github.com/luin/ioredis). Note: the [default parameters](https://github.com/luin/ioredis/blob/v4.16.0/API.md#new_Redis_new) of a redis connection are not the fastest to provide a rate-limit. We suggest to customize the connectTimeout and maxRetriesPerRequest as in the [example](https://github.com/fastify/fastify-rate-limit/tree/master/example/example.js).